src/transformers/adapters/model_mixin.py

```diff
@@ -469,6 +477,15 @@ def load_adapter_fusion(
                     main_load_name=load_name,
                     set_active=set_active,
                 )
+
+        loader = PredictionHeadLoader(self)
+        try:
+            loader.load(load_dir)
+        except ValueError:
+            logger.log(
+                msg="The loaded adapter fusion does not have a head. Before inference you should train or load a head.",
+                level=logging.INFO,
+            )
```

Comment:
Can't we do this similar to `load_adapter()`, i.e. add a custom weights loader in the ModelWithHeadsMixin and set a `with_head` flag there (see l. 759)?

Reply:
It should be done like that now.